oauthproxy.go

```diff
@@ -980,6 +983,10 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 
 // addHeadersForProxying adds the appropriate headers the request / response for proxying
 func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Request, session *sessionsapi.SessionState) {
+	for _, extraHeader := range p.extraHeaders {
+		match := headerSplitRegex.FindStringSubmatch(extraHeader)
+		req.Header[match[1]] = []string{match[2]}
+	}
```

Comment:
It would be better to split these into a map using the regex as part of the startup, rather than splitting every request

Reply:
Hmm... true.